### PR TITLE
Patient Summary and Charts

### PR DESCRIPTION
**Pull request by valvijo**
Including:
- [x] visit header
- [x] default location
- [x] patient flag service

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "angular-smart-table": "~2.1.1",
     "angular-chart.js": "~0.7.2",
     "angular-cookies": "~1.4.3",
-    "lodash": "4.6.1",
+    "lodash": "~4.6.1",
```
Comment (psbrandt) [inline bower.json]: The build is now failing on [Travis](https://travis-ci.org/esaude/esaude-emr-poc/builds/118979971#L724) with:

```sh
bower ECONFLICT     Unable to find suitable version for lodash
```

I can confirm that when I run `bower install` locally I get prompted to manually select a `lodash` version:

```sh
Unable to find a suitable version for lodash, please choose one:
    1) lodash#~3.10.1 which resolved to 3.10.1 and is required by angular-datepicker#1.0.20
    2) lodash#~4.6.1 which resolved to 4.6.1 and is required by ui-prototypes
    3) lodash#4.6.1 which resolved to 4.6.1
```
I can't commit to this PR, but you can fix the build on Travis by making the following change:

```diff
diff --git a/.travis.yml b/.travis.yml
index 6fb860d..10a2f1b 100755
--- a/.travis.yml
+++ b/.travis.yml
@@ -6,7 +6,7 @@ before_install:
     - sh -e /etc/init.d/xvfb start
 before_script:
     - npm install -g bower grunt-cli codacy-coverage coveralls
-    - bower install
+    - bower install --force-latest
 after_script:
     - cat coverage/**/lcov.info | codacy-coverage
     - cat coverage/**/lcov.info | coveralls
```

**Comment by valvijo** [comment]
Hi @psbrandt ,

I think the lodash problem is solved, but I'm having another issue when I try to run the tests locally, it says `TypeError: 'undefined' is not a function (evaluating 'controller('LoginController', {` and all the tests are failing, it looks like everything inside `inject(...)` is evaluating to `undefined`. Have you seen this before?

Thanks in advance

**Comment by psbrandt** [comment]
Hi @valvijo,

I see on Skype that you found the solution. You need to [configure Karma](http://karma-runner.github.io/0.13/config/files.html) to load all the scripts that you need for the tests into the browser.

I implemented the fix you posted on Skype locally, and it solved the `Module 'home' is not available` error, but now one of the login tests is failing because of [this change](https://github.com/esaude/esaude-emr-poc/pull/9/files#diff-d4793f7433afe48fb7a83b416e1f0526L17):

```sh
PhantomJS 2.1.1 (Mac OS X 0.0.0) Controller: LoginController should correctly set $scope.locations FAILED
	Expected undefined to equal [ Object({ uuid: '5c303ee9-ca10-43fc-829f-c9e45bb7748e', display: 'Angonia', links: [ Object({ rel: 'self', uri: 'NEED-TO-CONFIGURE/ws/rest/v1/location/5c303ee9-ca10-43fc-829f-c9e45bb7748e' }) ] }), Object({ uuid: '7fc3f286-15b1-465e-9013-b72916f58b2d', display: 'Cahora Bassa', links: [ Object({ rel: 'self', uri: 'NEED-TO-CONFIGURE/ws/rest/v1/location/7fc3f286-15b1-465e-9013-b72916f58b2d' }) ] }), Object({ uuid: '7863528a-35dc-453e-a671-73580ef78ba2', display: 'Changara', links: [ Object({ rel: 'self', uri: 'NEED-TO-CONFIGURE/ws/rest/v1/location/7863528a-35dc-453e-a671-73580ef78ba2' }) ] }) ].
```
I guess the test needs to be modified to check the new behaviour?

Cheers,
Pascal

Team eSaude

**Comment by valvijo** [comment]
Hi @psbrandt 

The problem is that I get [this](http://pastebin.com/RU8rF14c) error when I run the tests. To me this is like all the injected stuff are undefined. This must be something wrong with my local repo, once that you can run it, although assertions are failing. Will investigate and inform you latter.

Thanks
Valério

**Pull request merged**